manager/state/raft/util.go

```diff
@@ -26,6 +26,14 @@ func dial(addr string, protocol string, creds credentials.TransportAuthenticator
 	return grpc.Dial(addr, grpcOptions...)
 }
 
+// handshake pings a node back using its advertised address to check if it is
+// reachable or not.
+func handshake(addr string, creds credentials.TransportAuthenticator) error {
+	client, err := dial(addr, "tcp", creds, 3*time.Second)
```

Comment:
Isn't grpc's `Dial` async? I think we need some way to confirm that the connection succeeded.

Maybe we could force `Dial` to be synchronous in this context by passing that option.